inc/az_ulib_ustream_forward.h

```diff
@@ -0,0 +1,355 @@
+{
+  return ustream_forward->_internal.api->dispose(ustream_forward);
+}
+
+/**
+ * @brief   Factory to initialize a new ustream_forward.
+ *
+ *  This factory initializes a ustream_forward that handles the content of the provided buffer. As a
+ *      result, it will return an #az_ulib_ustream_forward* with this content. The initialized 
+ *      ustream_forward takes ownership of the passed memory and will release the memory with the 
+ *      passed #az_ulib_release_callback function when the caller goes out of scope.
+ *
+ * @param[out]      ustream_forward             The pointer to the allocated  
+ *                                              #az_ulib_ustream_forward struct. 
+ * @param[in]       ustream_forward_release     The #az_ulib_release_callback function that will be
+ *                                              called to release the ustream_forward block (the 
+ *                                              passed `ustream_forward` parameter). If `NULL` is 
+ *                                              passed, the `az_ulib_ustream_forward_dispose` will 
+ *                                              not release the memory associated with 
+ *                                              `ustream_forward`, so it will be the caller's 
+ *                                              responsibility to do so. For example, developers 
+ *                                              may use the stdlib `free` to release malloc'd 
+ *                                              memory.
+ * @param[in]       data_buffer                 The `const uint8_t* const` that points to a memory
+ *                                              position where the buffer starts.
+ * @param[in]       data_buffer_length          The `size_t` with the number of `uint8_t` in the
+ *                                              provided buffer.
+ * @param[in]       data_buffer_release         The #az_ulib_release_callback function that will be
+ *                                              called to release the data. If `NULL` is passed,
+ *                                              the `az_ulib_ustream_forward_dispose` will not  
+ *                                              release the memory associated with 
+ *                                              `ustream_forward`, so it will be the caller's 
+ *                                              responsibility to do so. For example, developers 
+ *                                              may use the stdlib `free` to release malloc'd 
+ *                                              memory.
+ * 
+ * @pre \p          ustream_forward             shall not be `NULL`.
+ * @pre \p          data_buffer                 shall not be `NULL`.
+ * @pre \p          data_buffer_length          shall be greater than `0`.
+ * 
+ * @return The #az_result with result of the initialization.
```

Comment:
>az_result [](http://example.com/codeflow?start=16&length=9)

For the APIs that return az_result, but currently we only know AZ_OK case, consider add a comment that it can return errors depending on the implementation.